toxcore/forwarding.h

```diff
@@ -0,0 +1,108 @@
+/* SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright © 2019-2022 The TokTok team.
+ */
+
+#ifndef C_TOXCORE_TOXCORE_FORWARDING_H
+#define C_TOXCORE_TOXCORE_FORWARDING_H
+
+#include "DHT.h"
+#include "network.h"
+
+#define SENDBACK_IPPORT 0
+#define SENDBACK_FORWARD 1
+#define SENDBACK_TCP 2
+
+#define MAX_SENDBACK_SIZE (0xff - 1)
+#define MAX_FORWARD_DATA_SIZE (MAX_UDP_PACKET_SIZE - (1 + 1 + MAX_SENDBACK_SIZE))
+
+#define MAX_FORWARD_CHAIN_LENGTH 4
+
+#define MAX_PACKED_IPPORT_SIZE (1 + SIZE_IP6 + sizeof(uint16_t))
+
+typedef struct Forwarding Forwarding;
+
+non_null()
+DHT *forwarding_get_dht(Forwarding *forwarding);
```

Comment:
This function seems to exist so you don't need to pass the DHT to the announce constructor. What was the reasoning here? It's certainly not less code, so are you trying to express some relationships or invariants through this extra code?

Reply:
It expresses that the dht object directly used for announce and that used by announce's forwarding object should be the same.

Reply:
SGTM.